### ISSUE TITLE
[Bug] T.tile.silu in-place (dst==src) silently produces wrong results on both ascendc and pto

### ISSUE THREAD

**Issue by fredrekelthen** [tile-ai/tilelang-ascend#1658]
## Bug Description

`T.tile.silu` 原地运算（dst == src 为同一 buffer）时静默产生错误结果，两后端均如此，无任何报错。已在 `request/说明和审查/框架.md` 第 12 条记录，根因已定位到 CANN 实现。

## Environment

**Hardware:**
- NPU Device: Ascend910B3（A2，dav-2201）

**Software:**
- OS: Linux
- CANN version: 8.5.2
- torch-npu version: 2.7.1
- Python version: 3.11
- TileLang-Ascend version: 0.1.4（refine-tile-relu-silu-sigmoid 分支）
- Installation method: build from source

## Code to Reproduce

```python
# Minimal reproducible code
import torch
import tilelang
import tilelang.language as T

PASS_CONFIGS = {
    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_COMBINE: True,
    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_SYNC: True,
    tilelang.PassConfigKey.TL_ASCEND_AUTO_SYNC: True,
    tilelang.PassConfigKey.TL_ASCEND_MEMORY_PLANNING: True,
}

@T.prim_func
def silu_inplace(
    A: T.Tensor((64, 64), "float16"),
    B: T.Tensor((64, 64), "float16"),
):
    with T.Kernel(1, is_npu=True) as (cid, vid):
        a_ub = T.alloc_ub((64, 64), "float16")
        T.copy(A, a_ub)
        T.tile.silu(a_ub, a_ub)  # in-place: dst == src
        T.copy(a_ub, B)

if __name__ == "__main__":
    a = torch.randn(64, 64, dtype=torch.float16, device="npu")
    for target in ["ascendc", "pto"]:
        compiled = tilelang.compile(silu_inplace, out_idx=[-1], pass_configs=PASS_CONFIGS, target=target)
        torch.npu.synchronize()
        b = compiled(a)
        torch.npu.synchronize()
        diff = (b - torch.nn.functional.silu(a)).abs().max().item()
        print(f"[silu-inplace] {target} max_diff={diff:.6f}")
```

## Error Message

无显式报错（静默错误）。实测数值：

```
[silu-inplace-2d-64x64] ascendc max_diff=2.730469
[silu-inplace-2d-64x64] pto max_diff=3.519531
（100% 元素不匹配）

非原地（独立 dst/src buffer）双后端均正确。
```

## Expected Behavior

`T.tile.silu(a_ub, a_ub)` 原地运算应输出与 `torch.nn.functional.silu(a)` 一致的结果（与 `T.tile.relu` / `T.tile.leaky_relu` 原地行为一致）；若不支持原地，应在编译期或运行时报错，而非静默产生错误结果。

## Actual Behavior

原地运算静默产生错误结果：64×64 float16 实测 ascendc max_diff≈2.7、pto ≈3.5，100% 元素不匹配，无任何报错。

**根因已定位**：CANN `silu_common_impl.h::SiluCalcSimplified` 的指令序列：

```cpp
Muls(dst, src, -1);         // step1: dst = -src        ← in-place 时 src 被覆盖
Exp(dst, dst);              // step2: dst = e^(-src)
Adds(dst, dst, 1.0);        // step3: dst = 1+e^(-src)
Div(dst, srcAddr, dstAddr); // step4: dst = src/(1+e^(-src))  ← 依赖原始 src，已被 step1 破坏
```

step4 的 `Div` 需要原始 src，但 step1 已将其覆盖为 `-src` → 数值错误。

**对照**：`sigmoid_impl.h::SigmoidIntrinsicsImpl` 最后一步为 `Div(dst, stackBuffer, dst)`，分母来自独立临时 buffer、不依赖原始 src，故 `T.tile.sigmoid` 原地在 ascendc 正确（pto 仍错误，见附注）。

## Additional Context

**Programming Mode:**
- [x] Developer mode (automatic sync/buffer management)
- [ ] Expert mode (manual T.Scope/T.barrier_all)
- [ ] Mixed mode

**Backend:**
- [x] Ascend C & PTO backend
- [ ] AscendNPU IR backend

**Pass Configs (if applicable):**
```python
pass_configs = {
    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_COMBINE: True,
    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_SYNC: True,
    tilelang.PassConfigKey.TL_ASCEND_AUTO_SYNC: True,
    tilelang.PassConfigKey.TL_ASCEND_MEMORY_PLANNING: True,
}
```

**Operator Type:**
激活函数（SiLU / element-wise unary）

**附注（关联问题）**：
- `T.tile.sigmoid` 原地：ascendc 正确（max_diff≈5e-4），pto 错误（max_diff≈0.98）——同样建议模板层对 dst==src 别名做拷贝保护或编译期拒绝。
- 建议修复方向：CANN 侧修复 `SiluCalcSimplified`（先拷贝 src 或重排为使用独立临时 buffer 的序列）；TileLang 模板层对齐名（dst==src）做拷贝保护或编译期拒绝。

**Additional Information:**
- Have you checked the [Programming Guide](../docs/TileLang-Ascend%20Programming%20Guide.md)? Yes
- Have you searched existing issues? Yes
- Any other context or screenshots? 详细记录见 PR #1657 对应更新说明与 `request/说明和审查/框架.md` 第 12 条
